projects/cashmere/src/lib/tabs/tab-set/tab-set.component.ts

```diff
@@ -499,29 +506,48 @@ export class TabSetComponent implements AfterContentInit {
     }
 
     private defaultToFirstRoute() {
-        const foundRoute = this._tabs
-            .map(tab => tab.routerLink)
-            .map(routerLink => this.mapRouterLinkToString(routerLink))
-            .find(routerLink => {
-                const currentRoute = this.router.url;
-                return currentRoute === routerLink || currentRoute.indexOf(`${routerLink}/`) > -1;
-            });
+        const tabArray = this._tabs.toArray();
+        let routeTab = -1;
+
+        // Determine if there is a routerLink in our tabset that matches the current route
+        for ( let i=0; i < tabArray.length; i++ ) {
+            const routerLink = this.mapRouterLinkToString(tabArray[i].routerLink);
+            const currentRoute = this.router.url.split("?")[0];
+            if( currentRoute === routerLink ) {
+                routeTab = i;
+                break;
+            }
+        }
 
-        if (foundRoute) {
+        // If there is a match, select that tab and make it active
+        if ( routeTab >= 0 ) {
+            this._selectedTab = tabArray[routeTab];
+            this._setActive( tabArray[routeTab] );
             return;
         }
 
-        const tabArray = this._tabs.toArray();
+        // If there isn't a match, navigate to the route of the default tab
         if (tabArray[Number(this.defaultTab)]) {
-            const firstRoute = this.mapRouterLinkToString(tabArray[Number(this.defaultTab)].routerLink);
-            this.router.navigate([firstRoute], {relativeTo: this.route});
+            const firstRouteLink = tabArray[Number(this.defaultTab)].routerLink;
+            const firstRouteArray = Array.isArray(firstRouteLink) ? firstRouteLink : [firstRouteLink];
+            this._selectedTab = tabArray[Number(this.defaultTab)];
+            this.router.navigate(firstRouteArray, {relativeTo: this.route, queryParams: tabArray[Number(this.defaultTab)].queryParams});
         }
     }
 
     private mapRouterLinkToString(routerLink: string | string[]): string {
+        // Combine url arrays
         if (routerLink instanceof Array) {
             routerLink = routerLink.join('/').replace('//', '/');
         }
+        // Resolve relative urls
+        if ( !routerLink.startsWith('/') ) {
+            const currentRoute = this.router.url.split("?")[0];
+            const routeParts = currentRoute.split('/');
+            routeParts.shift();
+            routeParts.pop();
```

Comment:
This bit of logic seems a little fragile. What if someone's using nested tabs?

Reply:
I _think_ it shouldn't matter because what I'm trying to do there is construct a (mostly) full path from a relative path.  So I'm grabbing the current url and stripping the first segment ("https://") and the last segment (which gets combined with the relative path).  So nested tabs with relative paths are still relative to the current url, no?

Reply:
`this.router.url` doesn't yield the same thing as `window.location.href`. For example, testing on one of the pages in PowerPlanning:

window.location.href: http://localhost:4300/budgeting/expenses/general-ledger
this.router.url: /budgeting/expenses/general-ledger

So once you shift and pop (the next great dance move in America's urban clubs, probably), you'd end up with `['expenses']`.

Imagining we had an outer tab set that navigates between "expenses" and its siblings (e.g. "volume", "spread") and an inner tab set that navigates between "general-ledger" and its siblings (e.g. "summary", "department"), would this method work as expected for both of them?

Reply:
I liked the shift and pop too 😆 Fortunately I was able to test this a bit last night because the Components section uses relative paths on the Examples, API, Usage tabs.  And because of that first `/` the first element of the split array was an empty string (hence the need for the shift).  So actually I think for PowerPlanning you'd get `budgeting/expenses` - which is what we want to put the relative path at the end of.

I'll try to set up the scenario you mentioned with inner and outer tabs to validate that.

Reply:
Oh, you're right. I messed up the mental calculation on that split.

Reply:
Have you looked into `ActivatedRoute.snapshot.url`? It's sensitive to the current component's position in the router tree, and returns only the route segment relevant to it.

Reply:
Good thought - I was thinking `ActivatedRoute` might be a better option than `Router` here.  I'll give the snapshot url a look.